### PR TITLE
feat/injection dry run

### PR DESCRIPTION
**Pull request by acharl**

### CODE REVIEW

File: src/app/services/copy/copy.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CopyService {
+  public copyToClipboard(val: string) {
+    // This would be a nicer solution, but it doesn't work in Safari
+    // (navigator as any).clipboard.writeText(val)
```
Comment (AndreasGassmann) [inline src/app/services/copy/copy.service.ts]: Is safari support necessary?

Also, I think in the beacon-sdk we use this or something similar and it also works in Safari.

File: src/extension/tezos-types.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import { TezosWrappedOperation } from '@airgap/coinlib-core'
+
+export type MetadataBalanceUpdatesKindEnum = 'contract' | 'freezer'
```
Comment (AndreasGassmann) [inline src/extension/tezos-types.ts]: Do we need all those types? Are they not in the coinlib?

File: src/app/pages/dry-run-preview/dry-run-preview.page.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  public jsonString: string | undefined
+  @Input()
+  public preapplyResponse: PreapplyResponse[] | undefined
+
+  constructor(private readonly modalController: ModalController) {}
+
+  ngOnInit(): void {
+    if (this.preapplyResponse) {
+      this.jsonString = JSON.stringify(this.preapplyResponse)
+      const operationResultErrors = this.preapplyResponse[0].contents.map(operation => {
```
Comment (AndreasGassmann) [inline src/app/pages/dry-run-preview/dry-run-preview.page.ts:21]: Is it ok to only take the first element of the response? Would it be possible, eg. through beacon, to have more operation groups?

File: src/extension/AirGapSigner.ts [hunk 4]
```diff
@@ -33,56 +35,128 @@ export class AirGapOperationProvider implements OperationProvider {
     return forgedTx.binaryTransaction
   }
 
+  public async operationGroupFromWrappedOperation(
```
Comment (AndreasGassmann) [inline src/extension/AirGapSigner.ts:38]: Would it make sense to move those methods to the coinlib so we can also use them in airgap?

File: src/app/services/copy/copy.service.ts (hunk 1 above)
Comment (godenzim) [reply src/app/services/copy/copy.service.ts]: we do not need to support safari for now

File: src/extension/tezos-types.ts (hunk 2 above)
Comment (godenzim) [reply src/extension/tezos-types.ts]: I reduced the types to the minimum necessary

File: src/app/pages/dry-run-preview/dry-run-preview.page.ts (hunk 3 above)
Comment (godenzim) [reply src/app/pages/dry-run-preview/dry-run-preview.page.ts:21]: For now it's ok

File: src/extension/AirGapSigner.ts (hunk 4 above)
Comment (godenzim) [reply src/extension/AirGapSigner.ts:38]: it's fine for now

**Review by godenzim — APPROVED** [review]

**Pull request merged**